pyproject.toml

```diff
@@ -10,32 +34,32 @@
 # There are some restrictions on what makes a valid project name
 # specification here:
 # https://packaging.python.org/specifications/core-metadata/#name
-name = "sampleproject"  # Required
+name = "sampleproject"  # REQUIRED, is the only field that cannot be marked as dynamic.
 
 # Versions should comply with PEP 440:
 # https://www.python.org/dev/peps/pep-0440/
 #
 # For a discussion on single-sourcing the version, see
 # https://packaging.python.org/guides/single-sourcing-package-version/
-version = "3.0.0"  # Required
+version = "3.0.0"  # REQUIRED, although can be dynamic
 
 # This is a one-line description or tagline of what your project does. This
 # corresponds to the "Summary" metadata field:
 # https://packaging.python.org/specifications/core-metadata/#summary
-description = "A sample Python project"  # Optional
+description = "A sample Python project"
 
 # This is an optional longer description of your project that represents
 # the body of text which users will see when they visit PyPI.
 #
 # Often, this is the same as your README, so you can just read it in from
-# that file directly (as we have already done above)
+# that file directly.
 #
 # This field corresponds to the "Description" metadata field:
 # https://packaging.python.org/specifications/core-metadata/#description-optional
-readme = "README.md" # Optional
```

Comment:
Same here — it's recommended but optional in practice.

Reply:
It's definitely worth it when publishing to PyPI (which should cover most cases).

Reply:
It is recommended, but PyPI does not reject uploads if it's missing.

I suppose the only practical way of knowing if something would be rejected is to run `twine check --strict` against a dist. That's your source of truth for what's really optional in the metadata. Note that backends may generate some metadata fields without consulting the configuration file (as a fallback, for instance).